RockLib.Collections/NamedCollection.cs

```diff
@@ -6,203 +6,208 @@
 
 namespace RockLib.Collections
 {
-    /// <summary>
-    /// Represents a collection that can be retrieved by name.
-    /// </summary>
-    /// <typeparam name="T">The type of items in the collection.</typeparam>
-    public class NamedCollection<T> : IReadOnlyCollection<T>, IReadOnlyDictionary<string, T>
-    {
-        private const string _defaultDefaultName = "default";
-
-        private readonly IReadOnlyDictionary<string, T> _valuesByName;
-        private readonly Func<T, string> _dictionaryKeySelector;
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="NamedCollection{T}"/> class
-        /// according to the specified <paramref name="getName"/> function.
-        /// </summary>
-        /// <param name="values">The values that make up the collection.</param>
-        /// <param name="getName">A function that gets the name of a value.</param>
-        /// <param name="stringComparer">
-        /// The equality comparer used to determine if names are equal. The default value is
-        /// <see cref="OrdinalIgnoreCase"/>.
-        /// </param>
-        /// <param name="defaultName">
-        /// The name that indicates an item should be used as the <see cref="DefaultValue"/>
-        /// for the <see cref="NamedCollection{T}"/>.
-        /// </param>
-        /// <param name="strict">
-        /// Whether to throw an exception if <paramref name="values"/> contains more than one
-        /// default value or more than one value with the same name. If <see langword="false"/>,
-        /// when there are duplicates, then the last value in the collection wins.
-        /// </param>
-        public NamedCollection(IEnumerable<T> values, Func<T, string> getName,
-            IEqualityComparer<string> stringComparer = null, string defaultName = _defaultDefaultName, bool strict = true)
-        {
-            if (values == null)
-                throw new ArgumentNullException(nameof(values));
-            if (getName == null)
-                throw new ArgumentNullException(nameof(getName));
-
-            StringComparer = stringComparer ?? OrdinalIgnoreCase;
-            DefaultName = !string.IsNullOrEmpty(defaultName) ? defaultName : _defaultDefaultName;
-
-            var valuesByName = new Dictionary<string, T>(StringComparer);
-            var hasDefaultValue = false;
-
-            foreach (var value in values)
+#pragma warning disable CA1033
+   /// <summary>
+   /// Represents a collection that can be retrieved by name.
+   /// </summary>
+   /// <typeparam name="T">The type of items in the collection.</typeparam>
+#pragma warning disable CA1710 // Identifiers should have correct suffix
```

Comment:
Haha ug.

Reply:
I'm OK with having a `#pragma warning` in now and then, especially for things that are already there and would be potentiall messy to change (as they might break people).

Reply:
Yeah I thought it was funny that it implemented two things that hit this.